csrc/fusion_segmenter.cpp

```diff
@@ -3903,15 +3903,8 @@ bool SegmentCandidateFinder::codeGenSupportedMerge(
   NVF_ERROR(
       areDirectlyConnected(group1, group2),
       "only support testing immediate producer-consumer groups");
-  if (options_.only_segment_resharding_exprs) {
-    for (auto group : {group1, group2}) {
-      for (auto expr : group->exprs()) {
-        if (isResharding(expr)) {
-          return false;
-        }
-      }
-    }
-    return true;
+  if (options_.custom_should_merge_groups != nullptr) {
```

Comment:
There's also segment_set that serves a similar purpose. But it's hard to tell why it's not sufficient without reviewing the following PRs.

Reply:
You are right that both mechanisms serve the same purpose. Btw, this was also true before this PR with the `only_segment_resharding_exprs` option. For the time being, besides the fact that passing a function is much closer to the existing code than moving to segmenter sets, I also find this way more lightweight and usable in this context, more precisely, it saves me 
1) a pass for adding the sets
2) adding an option in the segmenter to only segment according to the segmenter set
3) a pass for removing the segmenter sets